libraries/rush-lib/src/pluginFramework/RushLifeCycle.ts

```diff
@@ -86,16 +86,20 @@ export class RushLifecycleHooks {
   /**
    * The hook to run between preparing the common/temp folder and invoking the package manager during "rush install" or "rush update".
    */
-  public readonly beforeInstall: AsyncSeriesHook<[IGlobalCommand, Subspace]> = new AsyncSeriesHook<
-    [IGlobalCommand, Subspace]
-  >(['command', 'subspace'], 'beforeInstall');
+  public readonly beforeInstall: AsyncSeriesHook<[IGlobalCommand, Subspace, string | undefined]> =
+    new AsyncSeriesHook<[IGlobalCommand, Subspace, string | undefined]>(
```

Comment:
Note that we can use named tuples to make the function parameters clearer to consumers.